subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -1605,4 +1665,82 @@ public void configureConventionPlugin(@Nullable String comment, BlockStatement p
+                    return new BuildScriptBuilder.UpgradingHandler();
+                default:
+                    throw new IllegalStateException(String.format("Unknown handler: '%s'.", insecureProtocolOption));
+            }
+        }
+    }
+
+    public static class WarningHandler implements InsecureProtocolHandler {
+        private final BuildInitDsl dsl;
+
+        public WarningHandler(BuildInitDsl dsl) {
+            this.dsl = dsl;
+        }
+
+        @Override
+        public void handle(URL url, BuildScriptBuilder.ScriptBlockImpl statements) {
+            final boolean isGroovy = dsl.getId().equalsIgnoreCase("Groovy");
+            final String allowPrefix = isGroovy ? "a" : "isA";
+            final String fileExtension = isGroovy ? "gradle" : "gradle.kts";
+
+            final Logger logger = LoggerFactory.getLogger(BuildScriptBuilder.class);
+            logger.warn("If you wish to use this repository, you will have to uncomment the line '{}llowInsecureProtocol=true' in the generated build.{}} file.", allowPrefix, fileExtension);
+
+            statements.propertyAssignment(null, "url", new BuildScriptBuilder.MethodInvocationExpression(null, "uri", Collections.singletonList(new BuildScriptBuilder.StringValue(url.toString()))), true);
+            statements.comment(buildAllowInsecureProtocolComment(dsl));
+        }
+
+        private String buildAllowInsecureProtocolComment(BuildInitDsl dsl) {
+            final PropertyAssignment assignment = new PropertyAssignment(null, "allowInsecureProtocol", new BuildScriptBuilder.LiteralValue(true), true);
+
+            final StringWriter result = new StringWriter();
+            try (PrintWriter writer = new PrintWriter(result)) {
+                PrettyPrinter printer = new PrettyPrinter(syntaxFor(dsl), writer, false);
+                assignment.writeCodeTo(printer);
+                return result.toString();
+            } catch (Exception e) {
+                throw new GradleException("Could not write comment.", e);
+            }
+        }
+    }
+
+    public static class UpgradingHandler implements InsecureProtocolHandler {
+        @Override
+        public void handle(URL url, BuildScriptBuilder.ScriptBlockImpl statements) {
+            final Protocol protocol = Protocol.fromUrl(url);
+            if (protocol.hasReplacement()) {
+                final Logger logger = LoggerFactory.getLogger(BuildScriptBuilder.class);
+                logger.warn("Upgrading protocol to '{}'.", protocol.getReplacement());
+
+                final URL upgradedURL = Protocol.secureProtocol(url);
+                statements.propertyAssignment(null, "url", new BuildScriptBuilder.MethodInvocationExpression(null, "uri", Collections.singletonList(new BuildScriptBuilder.StringValue(upgradedURL.toString()))), true);
+            } else {
+                throw new GradleException(String.format("Can't upgrade insecure protocol: '%s' as no replacement exists.", protocol));
+            }
```

Comment:
Another way to replace `Protocol` with just `GUtil.isSecureUrl` and `GUtil.toSecureUrl`:

```suggestion

            final URI upgradedURL = GUtil.toSecureUrl(urlFromString(url));
            statements.propertyAssignment(null, "url", new BuildScriptBuilder.MethodInvocationExpression(null, "uri", Collections.singletonList(new BuildScriptBuilder.StringValue(upgradedURL.toString()))), true);
```